types/whoops/index.d.ts

```diff
@@ -0,0 +1,26 @@
+// Type definitions for whoops 4.1
+// Project: https://github.com/Kikobeats/whoops
+// Definitions by: Florian Imdahl <https://github.com/ffflorian>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.9
+
+declare namespace createErrorClass {
+    type ExtendError<E extends Error, P extends Record<string, any> = any> = (propsOrMessage?: P | string) => E & P;
+
+    function type(className?: string, props?: Record<string, any>): ExtendError<TypeError>;
+    function range(className?: string, props?: Record<string, any>): ExtendError<RangeError>;
+    // function eval(className?: string, props?: Record<string, any>): ExtendError<EvalError>;
```

Comment:
Does somebody have an idea how to declare this function? When I try declaring it like this I get the following error:

```
ERROR: 12:14  expect  TypeScript@next compile error: 
Invalid use of 'eval'. Modules are automatically in strict mode.
```